File: Lab Report 2/C++/Task3.cpp
#include<iostream>
using namespace std;

class IntegerSet {
    private:
        int set[50];

    public:
        IntegerSet();
        void NewIntegerSet(int arr[]);
        IntegerSet UnionOfIntegerSets(IntegerSet i);
        IntegerSet IntersectionOfIntegerSets(IntegerSet i);
        void InsertElement(int k);
        void DeleteElement(int m);
        void SetPrint();
        bool IsEqualTo(IntegerSet i);
};

IntegerSet::IntegerSet() {
    for(int i = 0; i < 50; i++) {
        set[i] = 0;
    }
}

void IntegerSet::NewIntegerSet(int arr[]){
    for(int i = 0; i<50; i++) {
        set[i] = arr[i];
    }
}

IntegerSet IntegerSet::UnionOfIntegerSets(IntegerSet i) {
    IntegerSet j;
    for (int n = 0; n < 50; n++) {
        j.set[n] = set[n] || i.set[n];
    }
    return j;
}

IntegerSet IntegerSet::IntersectionOfIntegerSets(IntegerSet i) {
    IntegerSet j;
    for (int n = 0; n < 50; n++) {
        j.set[n] = set[n] && i.set[n];
    }
    return j;
}

void IntegerSet::InsertElement(int k) {
    set[k] = 1;
}

void IntegerSet::DeleteElement(int m) {
    set[m] = 0;
}

void IntegerSet::SetPrint(){
    for (int i = 0; i < 50; i++) {
        cout<<set[i]<<" ";
    }
    cout<<endl;
}

bool IntegerSet::IsEqualTo(IntegerSet i){
    for (int n = 0; n < 50; n++) {
        if(set[n] != i.set[n]){
            return false;
        }
        return true;
    }
}

main() {
    int x[50],y[50];

    cout<<"Set values of Array X: ";
    for (int i = 0; i < 50; i++) {
        cin>>x[i];
        if(x[i] != 0) x[i] = 1;
    }

    cout<<"Set values of Array Y: ";
    for (int i = 0; i < 50; i++) {
        cin>>y[i];
        if(y[i] != 0) y[i] = 1;
    }

    IntegerSet i1,i2,i3;

    i1.NewIntegerSet(x);
    cout<<"I1: ";
    i1.SetPrint();

    i2.NewIntegerSet(y);
    cout<<"I2: ";
    i2.SetPrint();

    int k,m;

    cout<<"\nType location of insertion in I1: ";
    cin>>k;
    i1.InsertElement(k);

    cout<<"Type location of deletion in I1: ";
    cin>>m;
    i1.DeleteElement(m);

    cout<<endl;

    i3 = i1.UnionOfIntegerSets(i2);
    i3.SetPrint();

    i3 = i1.IntersectionOfIntegerSets(i2);
    i3.SetPrint();

    cout<<"\nAre i1 and i2 equal: "<<i1.IsEqualTo(i2);
}

File: Lab Report 2/Python/Task3.py
class IntegerSet:
    set = [0] * 50

    def __init__(self):
        pass

    def NewIntegerSet(self, arr: object):
        self.set = [arr[i] for i in range(50)]

    def UnionOfIntegerSets(self, i):
        j = IntegerSet()
        for n in range(50):
            j.set[n] = self.set[n] or i.set[n]
        return j

    def IntersectionOfIntegerSets(self, i):
        j = IntegerSet()
        for n in range(50):
            j.set[n] = self.set[n] and i.set[n]
        return j

    def InsertElement(self, k):
        self.set[k] = 1

    def DeleteElement(self, m):
        self.set[m] = 0

    def SetPrint(self):
        for i in self.set:
            print(self.set[i], end=" ")
        print()

    def IsEqualTo(self, i):
        if self.set == i.set:
            return True
        else:
            return False


# x = [0 for i in range(50)]
arr1 = input("Enter values for X: ").split(' ')
x = [int(i) for i in arr1]
print(x, len(x))

# y = [1 for i in range(50)]
arr2 = input("Enter values for Y: ").split(' ')
y = [int(i) for i in arr2]
print(y, len(y))

i1 = IntegerSet()
i2 = IntegerSet()
i3 = IntegerSet()

i1.NewIntegerSet(x)
print("I1:", end=" ")
i1.SetPrint()

i2.NewIntegerSet(y)
print("I2:", end=" ")
i2.SetPrint()

k = int(input("Location for insertion in I1: "))
i1.InsertElement(k)

m = int(input("Location for deletion from I1: "))
i1.DeleteElement(m)

i3 = i1.UnionOfIntegerSets(i2)
print("Union:", end=" ")
i3.SetPrint()

i3 = i1.UnionOfIntegerSets(i2)
print("Intersection:", end=" ")
i3.SetPrint()

print("Are I1 and I2 equal:", end=" ")
print(i1.IsEqualTo(i2));